web/src/message_lists.ts

```diff
@@ -61,12 +62,26 @@ export function set_current(msg_list: MessageList | undefined): void {
 }
 
 export function update_current_message_list(msg_list: MessageList | undefined): void {
-    if (current && !current.preserve_rendered_state) {
+    if (current && !current.should_preserve_current_rendered_state()) {
         // Remove the current message list from the DOM.
         current.view.$list.remove();
         rendered_message_lists.delete(current.id);
     } else {
+        // We plan to keep the current message list cached.
         current?.view.$list.removeClass("focused-message-list");
+        // Remove any existing message lists that we have with the same filter.
+        // TODO: If we start supporting more messages lists than just Combined feed,
+        // make this a proper filter comparison between the lists.
+        if (current?.data.filter.is_in_home()) {
+            for (const [id, msg_list] of rendered_message_lists) {
+                if (id !== current.id && msg_list.data.filter.is_in_home()) {
+                    msg_list.view.$list.remove();
+                    rendered_message_lists.delete(id);
+                    // We only expect to have one instance of a message list filter cached.
+                    break;
+                }
+            }
+        }
```

Comment:
Can you explain how this logic works? I can't quite tell how to reason about whether this might totally break the existing 

What I was thinking we could do is instead have narrowing to "combined feed" with a target message outside our rendered range work exactly like any other narrow with the filter equal to `is_in_home()` -- we don't use the existing rendered cache, since it doesn't contain the messages we need, and just ask the serve for messages around the anchor matching `in:home`.

Reply:
Started discussion in https://chat.zulip.org/#narrow/stream/6-frontend/topic/fetching.20message.20history/near/1797730.

The logic here checks if a message list with the same filter (aka combined feed) is already present, and if there is, we remove it from the rendered message list if we are planning to keep the current version of the message list with the same filter (aka combined feed) preserved.